soh/src/overlays/actors/ovl_En_Gm/z_en_gm.c

```diff
@@ -243,8 +247,17 @@ void EnGm_ProcessChoiceIndex(EnGm* this, GlobalContext* globalCtx) {
                     Message_ContinueTextbox(globalCtx, 0xC8);
                     this->actionFunc = func_80A3DD7C;
                 } else {
+                    if (gSaveContext.n64ddFlag && Randomizer_GetSettingValue(RSK_SHUFFLE_MERCHANTS) &&
+                        !Flags_GetRandomizerInf(RAND_INF_MERCHANTS_MEDIGORON)) {
+                            GiveItemEntryFromActor(&this->actor, globalCtx,
```

Comment:
So this actor check follows what I call the "fallback pattern" found in a handful of actor checks, in which the item give method is called in two places to account for edge cases. You can see below in func_80A3DF00 that it also calls the method you are replacing here, so you'll want to also do this condition and call there.
```
func_8002F434(&this->actor, globalCtx, GI_SWORD_KNIFE, 415.0f, 10.0f);
```

I know this is sorta confusing so let me know if it's not clear

Reply:
Just in case I may have misunderstood, do you mean something like this? (Sorry, for some reason Github doesn't like my codeblocks):
![image](https://user-images.githubusercontent.com/49068535/198409611-e0345791-5bdd-47d9-9e34-5106e836de99.png)
This was just copying what I did with `EnGm_ProcessChoiceIndex` onto `func_80A3DF00`.

Reply:
Yes, but IIRC you'd want the flag to only be set in the fallback? I can't recall what to do for flags in this situation cc @briaguya-ai

Reply:
do we have a good example to compare to? i know this one is a little different because we want to give the giant's knife instead of skipping the item give entirely, but i'm pretty sure there are still some good patterns sitting around

Reply:
@garrettjoecox could you elaborate a little on your concerns here? if this has been tested and seems to be working i'm willing to call it good

Reply:
Yea from what I can see it looks like some instances where there is a fallback pattern like this they are setting the flag before and in some instances they are setting it after.

It's still not really clear to me what all edge cases this fallback pattern handles outside of the shielding one (hold R before closing the final textbox before you receive the item) so as long as this case has been tested and everything else works I think we're good

Reply:
Just tested shielding before talking to Medigoron, and everything works as intended.

Reply:
Just to clarify, the important part is holding R (shield) while closing the last text box before you receive the item until after you've received the item.

Reply:
> Just to clarify, the important part is holding R (shield) while closing the last text box before you receive the item until after you've received the item.

Holding shield during the conversation with Medigoron does in fact give you the intended check item and not Giant's Knife, yes.